openeo/rest/datacube.py

```diff
@@ -523,34 +523,48 @@ def reduce_tiles_over_time(self, code: str, runtime="Python", version="latest"):
         """
         return self.reduce_temporal_udf(code=code, runtime=runtime, version=version)
 
-    def apply(self, process: str, data_argument='x') -> 'DataCube':
+    def apply(self, process: Union[PGNode, str], data_argument='x') -> 'DataCube':
         # TODO #125 allow more complex sub-process-graphs?
+
+        arguments = {"data": self._pg}
+
+        if self._api_version.at_least("1.0.0"):
+            if isinstance(process, str):
+                arguments["process"] = {"process_graph": PGNode(
+                        process_id=process,
+                        arguments={data_argument: {"from_parameter": "x"}})}
+            else:
+                arguments["process"] = {"process_graph": process}
+        else:
+            if isinstance(process, str):
+                arguments["process_graph"] = PGNode(
+                        process_id=process,
+                        arguments={data_argument: {"from_parameter": "x"}})
+            else:
+                arguments["process_graph"] = process
+
         return self.process_with_node(PGNode(
             process_id='apply',
-            arguments={
-                "data": self._pg,
-                "process": {"process_graph": PGNode(
-                    process_id=process,
-                    arguments={data_argument: {"from_parameter": "x"}}
-                )},
+            arguments=arguments
                 # TODO #125 context
-            }
         ))
 
-    def reduce_temporal_simple(self, process_id="max") -> 'DataCube':
+    def reduce_temporal_simple(self, process_id="max", dim_abbr="temporal") -> 'DataCube':
         """Do temporal reduce with a simple given process as callback."""
         return self._reduce_temporal(reducer=PGNode(
             process_id=process_id,
             arguments={"data": {"from_parameter": "data"}}
-        ))
+        ), dimension=dim_abbr)
 
-    def min_time(self) -> 'DataCube':
+    def min_time(self, dim_abbr='temporal') -> 'DataCube':
         """Finds the minimum value of a time series for all bands of the input dataset.
 
+            :param dim_abbr: Dimension name to reduce to.
+
             :return: a DataCube instance
         """
-
-        return self.reduce_temporal_simple("min")
+        # TODO: maybe find a better solution than dim_abbr (atm: time dimension: GEE: "t", VITO: "temporal")
+        return self.reduce_temporal_simple("min", dim_abbr=dim_abbr)
```

Comment:
see higher about `dimension` vs `dim_abbr`

also add comment `# TODO #116 determine dimension based on datacube metadata` here please (I'm planning to work on that somewhere this week)